### PR TITLE
build: add detect-secrets stage to build

### PR DESCRIPTION
**Pull request by padamstx**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,10 +20,25 @@ script:
...
+        - pip install --upgrade "git+https://github.com/ibm/detect-secrets.git@master#egg=detect-secrets"
+      script:
+        - detect-secrets scan --update .secrets.baseline
+        - detect-secrets -v audit --report --fail-on-unaudited --fail-on-live --fail-on-audited-real .secrets.baseline
+
+    - stage: Semantic-Release
+      language: node_js
+      node_js: 22
+      script:
+        - npm run semantic-release
```
Comment (dpopp07) [inline .travis.yml:44]: We don't have a `semantic-release` NPM target in this repo, which is why we were using `npx` before. So we need to either update this script to use `npx` or add a script target in the `package.json` file like in [other cores](https://github.com/IBM/go-sdk-core/blob/main/package.json#L16). I'm happy either way

**Review by dpopp07 — CHANGES_REQUESTED** [review]
Looks good but the semantic release step needs a fix

File: .travis.yml (hunk 1 above)
Comment (pyrooka) [reply .travis.yml:44]: I'm not the author, but nice catch. I don't think I would've been able to spot that... 😂

**Review by pyrooka — APPROVED** [review]
Besides that minor thing that Dustin found, it looks good.

File: .travis.yml (hunk 1 above)
Comment (padamstx) [reply .travis.yml:44]: I went ahead and added the script defn to package.json so that it's consistent with the other projects.

**Comment by padamstx** [comment]
> Looks good but the semantic release step needs a fix

Fixed

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**